### ISSUE TITLE
Placeholders always resolve against workspace root

### ISSUE THREAD

**Issue by memeplex** [foambubble/foam#864]
- Foam version: rev 4f116cfc8821852f6ac220296ba0cced1a7b4581
- Platform: Mac
- Issue occur on the [foam template](https://github.com/foambubble/foam-template) repo: Yes

**Summary**

Placeholders are using a different resolution sequence than regular wikilinks:

```ts
              const basedir =
                vscode.workspace.workspaceFolders.length > 0
                  ? fromVsCodeUri(vscode.workspace.workspaceFolders[0].uri)
                  : fromVsCodeUri(vscode.window.activeTextEditor?.document.uri)
                  ? URI.getDir(
                      fromVsCodeUri(
                        vscode.window.activeTextEditor!.document.uri
                      )
                    )
                  : undefined;
```

See https://github.com/foambubble/foam/blob/122db206956a36d145bea54041559a3b53e067a7/packages/foam-vscode/src/features/utility-commands.ts#L56.

Usually basedir will be `fromVsCodeUri(vscode.workspace.workspaceFolders[0].uri)` and relative paths will be relative to the workspace root (instead of the current file).

**Comment by riccardoferretti** [comment]
it's a feature, not a bug :)

jokes aside, I am guessing the underlying point of the issue is that you think a better default resolution for placeholders would be relative to the file where they are defined?

**Comment by memeplex** [comment]
There are different cases:
- Absolute path `/`:  it's fine now.
- Relative path `./`, `../`: it's wrong now because they should be anchored to the current file (bug).
- Identifier: this is ambiguous, I'd like to show a menu of all available directories (feature request).

**Comment by riccardoferretti** [comment]
For placeholders starting with`/` or `./` I believe your reasoning make sense (although personally I have never used them in that way, I can kinda see how someone might use that).

For the identifier case, I personally don't mind the current approach, but we could also provide a setting with the options of `root`, `currentDir` and `ask`. This is possibly connected to the #683, as this setting could actually be the "new note" directory

**Comment by faydanube** [comment]
> `/`对于以or开头的占位符，`./`我相信您的推理是有道理的（尽管我个人从未以这种方式使用过它们，但我可以看出有人会如何使用它）。
> 
> 对于标识符的情况，我个人不介意当前的方法，但我们也可以提供一个设置选项`root`,`currentDir`和`ask`。这可能与#683相关联，因为此设置实际上可能是“新笔记”目录

hi，VSCode Foam v0.20.7
// Specifies where to create a new note. It is overruled by the template or command arguments
//  - root: Use the root of the workspace
//  - currentDir: Use the directory of the file in the current editor
"foam.files.newNotePath": "currentDir",

This setting does not work anymore, it still works in the previous versions.
Because I have multiple directories, this setting suddenly failed to cause me distress

**Comment by riccardoferretti** [comment]
I am closing this issue as it has been fixed in 0.20.6, with the introduction of `foam.files.newNotePath`

Please file a new issue with bugs related to the feature
